types/command.js

```diff
@@ -124,12 +132,13 @@ class TypeCommand extends Type {
     if (context.commandHandlerRun) return this.resolve()
     context.commandHandlerRun = true
     this.api.addStrictModeErrors(context)
-    if (context.helpRequested || context.messages.length) {
-      // console.log('command.js postParse > adding deferred help, implicit:', match.implicit)
-      if (!context.output) context.addDeferredHelp(this.api.initHelpBuffer())
+    if (this._initPossibleHelpBuffer(context)) {
       return this.resolve()
     }
-    return this.runHandler(context.argv, context)
+    return Promise.resolve(this.runHandler(context.argv, context)).then(result => {
+      this._initPossibleHelpBuffer(context)
+      return result
+    })
```

Comment:
How about we use `await` here instead of `.then()`?

```suggestion
    const result = await this.runHandler(context.argv, context)
    this._initPossibleHelpBuffer(context)
    return result
```